### PR TITLE
Canvas use relation utils

### PR DESCRIPTION
**Pull request by huwshimi**
Replace the add and destroy relation code with the new utils. Working towards #1918.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5684/
Test PASSed.

File: jujugui/static/gui/src/app/utils/relation-utils.js [hunk 1]
```diff
@@ -518,29 +517,21 @@ YUI.add('relation-utils', function(Y) {
...
+  RelationUtils.createRelation = function(db, env, endpoints, callback) {
+    var endpointData = RelationUtils.parseEndpointStrings(
+      db, [endpoints[0][0], endpoints[1][0]]);
+    var match = RelationUtils.findEndpointMatch(endpointData);
+    var relationId = `pending-${endpoints[0][0]}:${endpoints[0][1].name}` +
+      `${endpoints[1][0]}:${endpoints[1][1].name}`;
     db.relations.add({
       relation_id: relationId,
-      'interface': endpoints[0][1].name,
+      interface: match.interface,
```
Comment (hatched) [inline jujugui/static/gui/src/app/utils/relation-utils.js]: `interface` is  a reserved word which is why it was in quotes.

File: karma-mocha-old.conf.js.tmpl [hunk 2]
```diff
@@ -10,7 +10,7 @@ module.exports = function(config) {
 
     // frameworks to use
     // available frameworks: https://npmjs.org/browse/keyword/karma-adapter
-    frameworks: ['mocha'],
+    frameworks: ['mocha', 'chai-sinon'],
```
Comment (hatched) [inline karma-mocha-old.conf.js.tmpl]: Good idea, thanks, we can now start using sinon in the old suite!

**Comment by huwshimi** [comment]
QA:
- add a couple of relatable services
- using the canvas try and create and then remove a relation
- using the inspector try and create and then remove a relation

**Comment by hatched** [comment]
This is a great improvement, thanks @huwshimi 

:+1: :ok:

File: karma-mocha-old.conf.js.tmpl (hunk 2 above)
Comment (huwshimi) [inline karma-mocha-old.conf.js.tmpl]: Yeah, I figured if we're adding new code here we might as well write it in a way that is compatible with the new suite.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5686/
Test PASSed.

**Comment by huwshimi** [comment]
Thanks @hatched :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**